src/Raven.Server/Documents/Queries/AST/Query.cs

```diff
@@ -447,20 +448,20 @@ public IEnumerable<object> GetFinalValues()
                 case AggregationType.Average:
                     for (int i = 0; i < _values.Count; i++)
                     {
-                        if ((long)_count[i] == 0)
+                        if (_count[i] == 0)
                         {
                             _values[i] = double.NaN;
                             continue;
                         }
 
-                        _values[i] = (double)_values[i] / (long)_count[i];
+                        _values[i] = _values[i] / _count[i];
                     }
                     break;
                 default:
                     throw new ArgumentOutOfRangeException("Unknown aggregation operation: " + Aggregation);
             }
 
-            return _values;
+            return _values.Cast<object>();
```

Comment:
I think those casting will box.. is it possible to get rid of the `object` and use `long` or `double` instead?

Reply:
we use those aggragated values in `AddTimeSeriesResult` :  https://github.com/ravendb/ravendb/blob/v5.0/src/Raven.Server/Documents/Queries/Results/TimeSeriesRetriever.cs#L828 .  we turn them into `DynamicJsonArray`..  and `DynamicJsonArray`  only works with objects